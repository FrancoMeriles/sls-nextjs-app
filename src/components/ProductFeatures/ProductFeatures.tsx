import { FC } from 'react'
import { Box, Grid, Flex, Divider, Text } from '@chakra-ui/react'
import { Check } from '@base/icons'

interface ProductFeaturesProps {
  attributes: AttributesProps
}

interface AttributesProps {
  map: any
  id: string
  name: string
  value_name: string
}

const ProductFeatures: FC<ProductFeaturesProps> = ({ attributes }) => {
  return (
    <>
      <Divider orientation="horizontal" />
      <Text fontSize="2xl" m="20px 0">
        Caracter√≠sticas del producto
      </Text>
      <Grid templateColumns="repeat(2, 1fr)" gap={4} rowGap={8} mt={12}>
        {attributes.map((attribute) => (
          <Flex key={attribute.id} alignItems="center">
            <Check boxSize={9} bg="brand.gray" borderRadius="50px" mr={2} />
            <Box>
              <Text fontSize="sm" mr={2}>
                {attribute.name}:
              </Text>
              <Text fontSize="sm" fontWeight="semibold">
                {attribute.value_name}
              </Text>
            </Box>
          </Flex>
        ))}
      </Grid>
    </>
  )
}

export default ProductFeatures
