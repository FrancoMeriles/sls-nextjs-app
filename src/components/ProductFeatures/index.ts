import ProductFeatures from './ProductFeatures'

export default ProductFeatures
