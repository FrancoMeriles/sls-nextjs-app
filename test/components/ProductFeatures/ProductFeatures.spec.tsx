import { render, screen } from '@testing-library/react'
import renderer from 'react-test-renderer'

import ProductFeatures from '@base/components/ProductFeatures'

const attributes = [
  {
    id: '__ID_1__',
    name: '__NAME_1__',
    value_name: '__NAME_1__',
  },
  {
    id: '__ID_2__',
    name: '__NAME_2__',
    value_name: '__NAME_2__',
  },
]
describe('ProductFeatures', () => {
  it('should render and match snapshot with required props', () => {
    const tree = renderer.create(<ProductFeatures attributes={attributes} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should renders and display the correct data', async () => {
    render(<ProductFeatures attributes={attributes} />)
    expect(screen.getByText('CaracterÃ­sticas del producto')).toBeInTheDocument()
    attributes.forEach((attribute) => {
      expect(screen.getByText(attribute.name)).toBeInTheDocument()
      expect(screen.getByText(attribute.value_name)).toBeInTheDocument()
    })
  })
})
