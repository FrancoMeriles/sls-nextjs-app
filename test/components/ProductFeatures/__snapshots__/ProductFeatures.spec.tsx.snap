// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductFeatures should render and match snapshot with required props 1`] = `
Array [
  <hr
    aria-orientation="horizontal"
    className="chakra-divider css-1upb9tn"
  />,
  <p
    className="chakra-text css-1hlj7v5"
  >
    Caracter√≠sticas del producto
  </p>,
  <div
    className="css-8rdblo"
  >
    <div
      className="css-70qvj9"
    >
      <svg
        className="chakra-icon css-1ypeky3"
        focusable={false}
        viewBox="0 0 32 32"
      >
        <path
          d="M16 8.85c1.593 0 3.064.521 4.253 1.402l-.962.91A5.851 5.851 0 1016 21.85a5.851 5.851 0 005.772-6.808l1.096-1.038A7.15 7.15 0 1116 8.85zm5.642 1.89l1.028 1.028-6.69 6.703-3.18-3.18 1.029-1.03 2.152 2.153 5.66-5.675z"
          fill="currentColor"
          fillRule="nonzero"
        />
      </svg>
      <div
        className="css-0"
      >
        <p
          className="chakra-text css-6ru19t"
        >
          __NAME_1__
          :
        </p>
        <p
          className="chakra-text css-17vgcs5"
        >
          __NAME_1__
        </p>
      </div>
    </div>
    <div
      className="css-70qvj9"
    >
      <svg
        className="chakra-icon css-1ypeky3"
        focusable={false}
        viewBox="0 0 32 32"
      >
        <path
          d="M16 8.85c1.593 0 3.064.521 4.253 1.402l-.962.91A5.851 5.851 0 1016 21.85a5.851 5.851 0 005.772-6.808l1.096-1.038A7.15 7.15 0 1116 8.85zm5.642 1.89l1.028 1.028-6.69 6.703-3.18-3.18 1.029-1.03 2.152 2.153 5.66-5.675z"
          fill="currentColor"
          fillRule="nonzero"
        />
      </svg>
      <div
        className="css-0"
      >
        <p
          className="chakra-text css-6ru19t"
        >
          __NAME_2__
          :
        </p>
        <p
          className="chakra-text css-17vgcs5"
        >
          __NAME_2__
        </p>
      </div>
    </div>
  </div>,
]
`;
